### PR TITLE
Add fixed point simulation interface

### PR DESCRIPTION
**Pull request by wswslzp**
Add simulation interface for fixed-point number type `UFix`, `SFix` and `AFix`. You can now read and write fixed-point signal directly.

### CODE REVIEW

File: core/src/main/scala/spinal/core/sim/package.scala [hunk 1]
```diff
@@ -435,6 +435,84 @@ package object sim {
...
+
+    def #= (that: BigDecimal): Unit = {
+      val rhs = (that * scala.math.pow(2, fractionLength)).toBigInt
+      require(rhs <= maxRawIntValue, s"$that is overflow. Max value allowed is $maxDouble")
+      require(rhs >= minRawIntValue, s"$that is underflow.Min value allowed is $minDouble")
+      bt.raw #= rhs
+    }
+    def #= (that : Double): Unit = this #= BigDecimal(that)
+    def randomize(): BigDecimal = {
+      var rhs = Random.nextDouble()
```
Comment (Dolu1990) [inline core/src/main/scala/spinal/core/sim/package.scala:505]: This is problematic, if the AFix / XFix are bigger than 52 bits, then the LSB will always be 0.
I think the best would be to randomize a BigInt value of the size of the "raw", and check it is in the bounds, else retry.

Comment (wswslzp) [reply core/src/main/scala/spinal/core/sim/package.scala:505]: I saw that this branch has been merged into @kleinai 's work and he has reworked the `randomize` part of AFix. For XFix part, I will fix it soon.

Comment (Dolu1990) [reply core/src/main/scala/spinal/core/sim/package.scala:505]: Ahhhh cool ^^, so i guess the klenai one will endup into SpinalHDL upstream too, so all good :)

**Comment by wswslzp** [comment]
Now that this branch has been merged into another branch, I will close this pr.

**Comment by Dolu1990** [comment]
Thanks anyway :)

**Pull request closed without merging**